src/err/mod.rs

```diff
@@ -800,15 +830,18 @@ impl PyErr {
     /// Return the cause (either an exception instance, or None, set by `raise ... from ...`)
     /// associated with the exception, as accessible from Python through `__cause__`.
     pub fn cause(&self, py: Python<'_>) -> Option<PyErr> {
-        let value = self.value(py);
-        let obj =
-            unsafe { py.from_owned_ptr_or_opt::<PyAny>(ffi::PyException_GetCause(value.as_ptr())) };
-        obj.map(Self::from_value)
+        let obj = unsafe {
+            Bound::from_owned_ptr_or_opt(
+                py,
+                ffi::PyException_GetCause(self.value_bound(py).as_ptr()),
+            )
+        };
+        obj.map(|inner| Self::from_value_bound(inner))
```

Comment:
I think we can golf this down a little:

```suggestion
        unsafe {
            ffi::PyException_GetCause(self.value_bound(py).as_ptr())
                .assume_owned_or_opt(py)
        }.map(Self::from_value_bound)
```

`rustfmt` probably doesn't like what I've written though 😂